lib/internal/modules/esm/hooks.js

```diff
@@ -553,15 +569,30 @@ class HooksProxy {
     }
   }
 
-  async makeAsyncRequest(method, ...args) {
+  /**
+   * Invoke a remote method asynchronously.
+   * @param {any[]|undefined} transferList Objects in `args` to be transferred
+   * @param {string} method Method to invoke
+   * @param  {any[]} args Arguments to pass to `method`
+   * @returns {Promise<any>}
+   */
+  async makeAsyncRequest(transferList, method, ...args) {
```

Comment:
Using variable arguments is probably more efficient than using arrays, because it's an internal method it would make sense to me to favor performance. It's easy to change after the fact anyway.

Reply:
Okay so then to make `method` first would it need to be `makeAsyncRequest(method, transferList, ...args)`, and `transferList` needs to be specified (even if just as `undefined`)? Or `makeAsyncRequest(method, ...args)` and we pull `transferList` out of the first element of `args`?

Reply:
It's a bit ugly looking I agree… but this was the only sensible way to preserve the `"command", arg1, arg2` style syntax that I saw.